internal/client/server.go

```diff
@@ -38,8 +38,8 @@ func (c *Project) initServerClient(ctx context.Context, cfg *config) (*grpc.Clie
 
 	// If we're local, then connection is optional.
 	opts := cfg.connectOpts
-	if c.local {
-		log.Trace("WithLocal set, server credentials optional")
+	if !c.noLocalServer {
+		log.Trace("Local server may be created later - server credentials optional")
```

Comment:
I'm still not entirely clear what I should expect my server's status to be. Maybe something slightly more specific?
```suggestion
		log.Trace("In-memory server specified; local server may be created later - server credentials optional")
```

Reply:
haha, it's actually not entirely clear at this point - I believe our logic is that if we don't detect a context with a server address, and `noLocalServer` isn't set, we'll create a local server.

So at this point, we don't yet know if we'll end up creating an in-memory server, we just know that it _may_ happen.